lib/common/iso8601.c

```diff
@@ -97,6 +99,30 @@ crm_time_new(const char *date_time)
     return dt;
 }
 
+static void ha_set_tm_time(crm_time_t * target, struct tm *source);
+
+crm_time_t *
+crm_time_new_lang_aware(const char *date_time)
+{
+    struct tm tm_spec;
+    crm_time_t *dt = NULL;
+
+    tzset();
+#ifdef D_T_FMT
+    if (strptime(date_time, nl_langinfo(D_T_FMT), &tm_spec) != NULL
+#ifdef _DATE_FMT
+    /* _DATE_FMT is what coreutils' date uses internally */
+            || strptime(date_time, nl_langinfo(_DATE_FMT), &tm_spec) != NULL
+#endif
+    ) {
```

Comment:
strptime() doesn't accept a NULL date_time, so handle that first

Reply:
I see, it happened accidentally, the order was flipped origially
(so the NULL check was handled as a side effect of that), but I
had to change that because of overly verbose complaints when it
didn't match the required strict format and didn't feel like
fighting with the system at the moment.

I know see that Gnulib needs the same treatment, good that you
spotted that, thanks.